tinyDA/distributions.py

```diff
@@ -278,6 +290,12 @@ def __init__(self, data, covariance):
         self.data = data
         self.cov = np.diag(covariance)
 
+        if not isinstance(covariance, np.ndarray):
+            raise TypeError("Covariance must be a numpy array.")
+        elif covariance.ndim == 1:
+            if not covariance.shape[0] == data.shape[0]:
+                raise ValueError("Dimensions of data and covariance do not match.")
+
```

Comment:
Do we need these checks for `DiagonalGaussianLogLike`? This is already checked by the upper-level function `GaussianLogLike`.